packages/solid/src/reactive/signal.ts

```diff
@@ -189,9 +200,22 @@ export function createSignal<T>(
  *
  * @description https://www.solidjs.com/docs/latest/api#createcomputed
  */
-export function createComputed<T>(fn: (v?: T) => T | undefined): void;
-export function createComputed<T>(fn: (v: T) => T, value: T, options?: { name?: string }): void;
-export function createComputed<T>(fn: (v?: T) => T, value?: T, options?: { name?: string }): void {
+export function createComputed<T extends U, U = T>(fn: EffectFunction<T, U>): void;
```

Comment:
Noted about the test case you mentioned to add. :+1: 

> `createComputed((v: number) => 123)` should not be deemed correct (since `v` is undefined on first execution). I think the effect functions should drop this first overload and `value` should be optional in the second overload i.e.:
> 
> ```ts
> // export function createComputed<T extends U, U = T>(fn: EffectFunction<T, U>): void; remove
> export function createComputed<T extends U, U = T>(
>   fn: EffectFunction<T, U | undefined>,
>   value?: undefined, // make optional
>   options?: EffectOptions
> ): void;
> export function createComputed<T extends U, U = T>(
>   fn: EffectFunction<T, U>,
>   value: U extends U ? U : never,
>   options?: EffectOptions
> ): void;
> ```
> 
> The third overload handles the `U = undefined` case as well, but this might be less confusing.
> 
> I'm unsure the second generic is needed for the first two overloads, are they for usability? `EffectFunction<T, U | undefined>` always returns T and can only be supplied with `T | undefined`, so `U` is redundant otherwise.

I was aiming for things like `(v: number | string) => number`, where `U` is `number | string`. Or is that different what you're talking about?